tests/certs/scripts/genserv.sh

```diff
@@ -59,6 +59,15 @@ if [ ".$SERIAL" = . ] ; then
 	SERIAL=`/usr/bin/env perl -e "$GETSERIAL"`
 fi
 
+# A hack for the expired certificate test case. We could do better, but it would
+# require using "openssl ca" instead of "openssl x509" for creating the server
+# certificates, as "openssl ca" lets you precisely specify start and end dates.
+case "$PREFIX" in
+*exp)
+    DURATION=-$DURATION
+    ;;
+esac
```

Comment:
This is very unusual, such certificates should not be encountered in the real world. I can imagine that some implementations refuse such certificates.

Would it be possible to explore the `openssl ca` approach to have more realistic certificates?